src/components/expression/expression-renderer.js

```diff
@@ -452,6 +513,75 @@ genotet.ExpressionRenderer.prototype.drawMatrixConditionLabels_ = function() {
 };
 
 /**
+ * Renders the expression heatmap gradient.
+ * @private
+ */
+genotet.ExpressionRenderer.prototype.drawHeatmapGradient_ = function() {
+  var heatmapData = this.data.matrix;
+  var scaleMin = 0;
+  var scaleMax = 0;
+  if (this.data.options.autoScaleGradient) {
+    scaleMin = heatmapData.valueMin;
+    scaleMax = heatmapData.valueMax;
+  } else {
+    scaleMin = heatmapData.allValueMin;
+    scaleMax = heatmapData.allValueMax;
+  }
+  scaleMin = scaleMin.toFixed(2);
+  scaleMax = scaleMax.toFixed(2);
+  var marginLeft = scaleMin.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR;
+  var marginRight = scaleMax.toString().length *
+    this.CONDITION_LABEL_HEIGHT_FACTOR;
+
+  var transformLeft = 0;
+  var transformTop = this.heatmapHeight_;
+  if (this.data.options.showProfiles) {
+    transformTop += this.profileHeight_;
+  }
+  if (this.data.options.showGeneLabels) {
+    transformLeft += this.geneLabelWidth_ + this.LABEL_MARGIN;
+  }
+  else if (this.data.options.showProfiles) {
+    transformLeft += this.DEFAULT_PROFILE_MARGIN;
+  }
+  if (this.data.options.showConditionLabels) {
+    transformTop += this.conditionLabelHeight_ +
+      this.LABEL_MARGIN + this.LABEL_DIFFERENCE;
+  }
+  this.svgHeatmapGradient_
+    .attr('transform',
+      genotet.utils.getTransform([transformLeft, transformTop]));
+  var gradientHeight = this.DEFAULT_HEATMAP_GRADIENT_HEIGHT -
+    this.heatmapGradientMargins_.TOP -
+    this.heatmapGradientMargins_.BOTTOM;
+  var gradientWidth = this.heatmapWidth_ - marginLeft - marginRight;
+  this.gradientRect_
+    .attr('transform', genotet.utils.getTransform([
+      marginLeft,
+      this.heatmapGradientMargins_.TOP
+    ]))
+    .attr('width', gradientWidth)
+    .attr('height', gradientHeight)
+    .attr('rx', 10)
+    .attr('ry', 10);
+  d3.select('.gradient-text-left')
+    .attr('x', marginLeft - this.HEATMAP_GRADIENT_MARGIN)
+    .attr('y',
```

Comment:
```js
.attr('y', this.heatmapGradientMargins_.TOP + gradientHeight / 2 +
   this.TEXT_HEIGHT / 3)
```
Try to put as many operators in a line as possible, unless it is much clearer to start a new line, e.g. for constant array declaration.